src/lib/i18n-localeParser.js

```diff
@@ -0,0 +1,46 @@
+/**
+ *
+ * @param {object} options
+ * @param {import('i18next').i18n} options.i18n
+ * @param {string} options.localeCode
+ * @param {Record<string, { locale: string, nativeName: string, englishName: string }>} options.languages
+ *
+ * @returns {string}
+ */
+export default function getValidLocaleCode ({ i18n, localeCode, languages }) {
```

Comment:
before creating this, I created `src/lib/i18n.test.js`. Then I created this file, added tests for it, and made sure the existing `src/lib/i18n.test.js` tests were passing.